trie.c

```diff
@@ -9,15 +9,19 @@
 #include <stdint.h>
 #include <string.h>
 #include <stdbool.h>
+#include <ctype.h>
 #include "keybuffer.h"
+#include "key_stack.h"
 #include "trie.h"
+#include "utils.h"
 #include "print.h"
 
 #define TRIE_MATCH_BIT      0x8000
 #define TRIE_BRANCH_BIT     0x4000
 #define TRIE_CODE_MASK      0x3FFF
 
 #define TDATA(L) pgm_read_word(&trie->data[L])
+#define CDATA(L) pgm_read_byte(&trie->completions[L])
```

Comment:
We already had `CDATA` in `sequence_transform.c`, so maybe should move it out somewhere

Reply:
I believe that this can't be in a header, unless we want to pollute the global namespace.

Reply:
Yes, and I also don't want these in a header cause they are very function specific (assume the definition of a `trie` pointer).
For now it's OK if multiple .c files reuse these I think.